pkg/kubelet/checkpointmanager/errors/errors.go

```diff
@@ -18,8 +18,18 @@ package errors
 
 import "fmt"
 
-// ErrCorruptCheckpoint error is reported when checksum does not match
-var ErrCorruptCheckpoint = fmt.Errorf("checkpoint is corrupted")
+// ErrCorruptCheckpoint error is reported when checksum does not match.
+// Check for it with:
+//
+//	var csErr CorruptCheckpointError
+//	if errors.As(err, &csErr) { ... }
+type CorruptCheckpointError struct {
+	ActualCS, ExpectedCS uint64
+}
+
+func (err CorruptCheckpointError) Error() string {
+	return "checkpoint is corrupted"
+}
```

Comment:
Is this only done so that you can see the difference in the checksum to put together the test cases?

Reply:
Yes. They are not shown to users.

Reply:
I was hoping that `assert.NoError` would dump the full error struct, like Gomega does. Unfortunately it doesn't, so one has to use `dlv` to print the error after setting a breakpoint on the line with the failing assert.

Reply:
https://github.com/kubernetes/kubernetes/pull/126156 explicitly prints the expected and actual checksum by peeking into the error.